src/formatDate.js

```diff
@@ -8,7 +8,26 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+  const dateParts = date.split(fromSeparator);
+  const dateMap = fromFormat.reduce((map, part, index) => {
+    map[part] = dateParts[index];
+
+    return map;
+  }, {});
+
+  if (dateMap['YYYY'] && toFormat.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
+  } else if (dateMap['YY'] && toFormat.includes('YYYY')) {
+    const year = parseInt(dateMap['YY'], 10);
+
+    dateMap['YYYY'] = year < 30 ? `20${dateMap['YY']}` : `19${dateMap['YY']}`;
```

Comment:
The condition to check for 'YYYY' in `dateMap` and 'YY' in `toFormat` is not comprehensive. If 'YYYY' exists in `dateMap` but 'YY' is not part of `toFormat`, the conversion is unnecessary and should not be performed. Additionally, when checking for 'YY' in `dateMap`, you should also verify that 'YYYY' is part of `toFormat` before performing the conversion.